### PR TITLE
oxefmsynth: init at 1.3.5

### PR DESCRIPTION
**Pull request by hirenashah**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/oxefmsynth/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{ stdenv, fetchFromGitHub, unzip, requireFile, libX11}:
+
+stdenv.mkDerivation rec {
+  pname = "oxefmsynth";
+  version = "v1.3.5";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/oxefmsynth/default.nix:5]:
```suggestion
  version = "1.3.5";
```

File: pkgs/applications/audio/oxefmsynth/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  src = fetchFromGitHub {
+    owner = "oxesoft";
+    repo = pname;
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/oxefmsynth/default.nix:10]:
```suggestion
    rev = "v{version}";
```

File: pkgs/applications/audio/oxefmsynth/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    sha256 = "1rk71ls33a38wx8i22plsi7d89cqqxrfxknq5i4f9igsw1ipm4gn";
+  };
+
+  vst-sdk = stdenv.mkDerivation rec {
+    name = "vstsdk368_08_11_2017_build_121";
+    src = requireFile {
+      name = "${name}.zip";
+      url = "http://www.steinberg.net/en/company/developers.html";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/oxefmsynth/default.nix:18]: That is a html url not a zip. Also please move this into a let in block.

File: pkgs/applications/audio/oxefmsynth/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      sha256 = "e0f235d8826d70f1ae0ae5929cd198acae1ecff74612fde5c60cbfb45c2f4a70";
+    };
+    nativeBuildInputs = [ unzip ];
+    installPhase = "cp -r . $out";
+    meta.license = stdenv.lib.licenses.unfree;
+  };
+
+  NIX_CFLAGS_COMPILE = [ "-Wno-error=narrowing" ];
+
+  buildFlags = "VSTSDK_PATH=${vst-sdk}/VST2_SDK";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/oxefmsynth/default.nix:28]:
```suggestion
  buildFlags = [ "VSTSDK_PATH=${vst-sdk}/VST2_SDK" ];
```

File: pkgs/applications/audio/oxefmsynth/default.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    mkdir -p $out/lib/lxvst
+    cp oxevst64.so $out/lib/lxvst
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/oxesoft/oxefmsynth";
+    description = "An open source VST 2.4 instrument plugin.";
+    maintainers = [ maintainers.hirenashah ];
+    platforms = platforms.linux;
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/oxefmsynth/default.nix:44]: Plus or Only?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/oxefmsynth/default.nix (hunk 3 above)
Comment (hirenashah) [reply pkgs/applications/audio/oxefmsynth/default.nix]: Yes, I took it from airwave package. Steinberg is not hosting the zip anymore. I updated the url to be a copy of it that I found in the Internet Archive.

File: pkgs/applications/audio/oxefmsynth/default.nix (hunk 5 above)
Comment (hirenashah) [reply pkgs/applications/audio/oxefmsynth/default.nix]: Fixed.

File: pkgs/applications/audio/oxefmsynth/default.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
+{ stdenv, fetchFromGitHub, unzip, requireFile, libX11 }:
+
+let
+
+  vst-sdk = stdenv.mkDerivation rec {
+    name = "vstsdk3610_11_06_2018_build_37";
+    src = requireFile {
+      name = "${name}.zip";
+      url =
+        "https://web.archive.org/web/20181016150224if_/https://download.steinberg.net/sdk_downloads/vstsdk3610_11_06_2018_build_37.zip";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/oxefmsynth/default.nix:10]:
```suggestion
      url = "https://web.archive.org/web/20181016150224if_/https://download.steinberg.net/sdk_downloads/vstsdk3610_11_06_2018_build_37.zip";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/oxefmsynth/default.nix [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  buildInputs = [ libX11 ];
+
+  installPhase = ''
+    mkdir -p $out/lib/lxvst
+    cp oxevst64.so $out/lib/lxvst
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/oxesoft/oxefmsynth";
+    description = "An open source VST 2.4 instrument plugin.";
```
Comment (sikmir) [inline pkgs/applications/audio/oxefmsynth/default.nix:40]:
```suggestion
    description = "An open source VST 2.4 instrument plugin";
```

File: pkgs/applications/audio/oxefmsynth/default.nix [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+  NIX_CFLAGS_COMPILE = [ "-Wno-error=narrowing" ];
+
+  buildFlags = [ "VSTSDK_PATH=${vst-sdk}/VST2_SDK" ];
+
+  buildInputs = [ libX11 ];
+
+  installPhase = ''
+    mkdir -p $out/lib/lxvst
+    cp oxevst64.so $out/lib/lxvst
```
Comment (sikmir) [inline pkgs/applications/audio/oxefmsynth/default.nix:35]:
```suggestion
    install -Dm644 oxevst64.so -t $out/lib/lxvst
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108859` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>oxefmsynth: log was empty</li>
  </ul>
</details>

File: pkgs/applications/audio/oxefmsynth/default.nix [hunk 9]
```diff
@@ -0,0 +1,45 @@
+{ stdenv, fetchFromGitHub, unzip, requireFile, libX11 }:
+
+let
+
+  vst-sdk = stdenv.mkDerivation rec {
+    name = "vstsdk3610_11_06_2018_build_37";
+    src = requireFile {
+      name = "${name}.zip";
+      url = "https://web.archive.org/web/20181016150224if_/https://download.steinberg.net/sdk_downloads/vstsdk3610_11_06_2018_build_37.zip";
```
Comment (rmcgibbo) [inline pkgs/applications/audio/oxefmsynth/default.nix:9]: Aparently this cannot be downloaded automatically?

```
trace: Warning: `stdenv.lib` is deprecated and will be removed in the next release. Please use `lib` instead. For more information see https://github.com/NixOS/nixpkgs/issues/108938
these 4 derivations will be built:
  /nix/store/mza3b9ang78j06axzn9w83hrliq2rbmy-vstsdk3610_11_06_2018_build_37.zip.drv
  /nix/store/8apzlm7ksgaziwgjjcqxf6grk9cks9q4-vstsdk3610_11_06_2018_build_37.drv
  /nix/store/nwbspwi173fibnmdqxdpkdfq4hfnws73-oxefmsynth-1.3.5.drv
  /nix/store/iyxkazn05qzls3ca8izvz1smk7p3f8c3-review-shell.drv
building '/nix/store/mza3b9ang78j06axzn9w83hrliq2rbmy-vstsdk3610_11_06_2018_build_37.zip.drv'...


***
***
Unfortunately, we cannot download file vstsdk3610_11_06_2018_build_37.zip automatically.
Unfortunately, we cannot download file vstsdk3610_11_06_2018_build_37.zip automatically.
Please go to https://web.archive.org/web/20181016150224if_/https://download.steinberg.net/sdk_downloads/vstsdk3610_11_06_2018_build_37.zip to download it yourself, and add it to the Nix store
Please go to https://web.archive.org/web/20181016150224if_/https://download.steinberg.net/sdk_downloads/vstsdk3610_11_06_2018_build_37.zip to download it yourself, and add it to the Nix store
using either
using either
  nix-store --add-fixed sha256 vstsdk3610_11_06_2018_build_37.zip
  nix-store --add-fixed sha256 vstsdk3610_11_06_2018_build_37.zip
or
or
  nix-prefetch-url --type sha256 file:///path/to/vstsdk3610_11_06_2018_build_37.zip
  nix-prefetch-url --type sha256 file:///path/to/vstsdk3610_11_06_2018_build_37.zip


***
***


builder for '/nix/store/mza3b9ang78j06axzn9w83hrliq2rbmy-vstsdk3610_11_06_2018_build_37.zip.drv' failed with exit code 1
cannot build derivation '/nix/store/8apzlm7ksgaziwgjjcqxf6grk9cks9q4-vstsdk3610_11_06_2018_build_37.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/nwbspwi173fibnmdqxdpkdfq4hfnws73-oxefmsynth-1.3.5.drv': 1 dependencies couldn't be built
```

File: pkgs/applications/audio/oxefmsynth/default.nix [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  buildFlags = [ "VSTSDK_PATH=${vst-sdk}/VST2_SDK" ];
+
+  buildInputs = [ libX11 ];
+
+  installPhase = ''
+    mkdir -p $out/lib/lxvst
+    install -Dm644 oxevst64.so -t $out/lib/lxvst
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/oxefmsynth/default.nix:38]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/applications/audio/oxefmsynth/default.nix [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+  vst-sdk = stdenv.mkDerivation rec {
+    name = "vstsdk3610_11_06_2018_build_37";
+    src = fetchurl {
+      url =
+        "https://web.archive.org/web/20181016150224if_/https://download.steinberg.net/sdk_downloads/${name}.zip";
+      sha256 = "176lr4gadgdnvv56q19amf533g5wbgvq4s28fi0zldjnjl1iwp9y";
+    };
+    nativeBuildInputs = [ unzip ];
+    installPhase = "cp -r . $out";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/oxefmsynth/default.nix:13]:
```suggestion
    name = "vstsdk3610_11_06_2018_build_37";
    src = fetchzip {
      url = "https://web.archive.org/web/20181016150224if_/https://download.steinberg.net/sdk_downloads/${name}.zip";
      sha256 = "176lr4gadgdnvv56q19amf533g5wbgvq4s28fi0zldjnjl1iwp9y";
    };
```

File: pkgs/applications/audio/oxefmsynth/default.nix [hunk 12]
```diff
@@ -0,0 +1,42 @@
+{ stdenv, lib, fetchFromGitHub, unzip, fetchzip, libX11 }:
+
+let
+
+  vst-sdk = stdenv.mkDerivation rec {
+    name = "vstsdk3610_11_06_2018_build_37";
+    src = fetchzip {
+      url = "https://web.archive.org/web/20181016150224if_/https://download.steinberg.net/sdk_downloads/${name}.zip";
+      sha256 = "176lr4gadgdnvv56q19amf533g5wbgvq4s28fi0zldjnjl1iwp9y";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/oxefmsynth/default.nix:9]:
```suggestion
      sha256 = "0da16iwac590wphz2sm5afrfj42jrsnkr1bxcy93lj7a369ildkj";
```

**Comment by SuperSandro2000** [comment]
@hirenashah FTBFS with

```
make: *** No rule to make target 'install'.  Stop.
```

File: pkgs/applications/audio/oxefmsynth/default.nix [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+  installPhase = ''
+    mkdir -p $out/lib/lxvst
+    install -Dm644 oxevst64.so -t $out/lib/lxvst
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/oxesoft/oxefmsynth";
+    description = "An open source VST 2.4 instrument plugin";
+    maintainers = [ maintainers.hirenashah ];
+    platforms = intersectLists platforms.x86_64 platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/oxefmsynth/default.nix:40]:
```suggestion
    platforms = [ "x86_64-linux" ];
```

**Pull request merged**